js/coinbase.js

```diff
@@ -1093,33 +1088,79 @@ module.exports = class coinbase extends Exchange {
         return this.parseTicker ([ spot, buy, sell ], market);
     }
 
-    parseTicker (ticker, market = undefined) {
+    async fetchTickerV3 (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'product_id': market['id'],
+            'limit': 1,
+        };
+        const response = await this.v3PrivateGetBrokerageProductsProductIdTicker (this.extend (request, params));
         //
-        // fetchTicker
+        //     {
+        //         "trades": [
+        //             {
+        //                 "trade_id": "10209805",
+        //                 "product_id": "BTC-USDT",
+        //                 "price": "19381.27",
+        //                 "size": "0.1",
+        //                 "time": "2023-01-13T20:35:41.865970Z",
+        //                 "side": "BUY",
+        //                 "bid": "",
+        //                 "ask": ""
+        //             }
+        //         ]
+        //     }
+        //
+        const data = this.safeValue (response, 'trades', []);
+        return this.parseTicker (data[0], market);
+    }
+
+    async parseTicker (ticker, market = undefined) {
+        //
+        // fetchTickerV2
         //
         //     [
         //         "48691.23", // spot
         //         "48691.23", // buy
         //         "48691.23",  // sell
         //     ]
         //
-        // fetchTickers
+        // fetchTickerV3
         //
-        //     "48691.23"
+        //     {
+        //         "trade_id": "10209805",
+        //         "product_id": "BTC-USDT",
+        //         "price": "19381.27",
+        //         "size": "0.1",
+        //         "time": "2023-01-13T20:35:41.865970Z",
+        //         "side": "BUY",
+        //         "bid": "",
+        //         "ask": ""
+        //     }
         //
+        const fetchTickerMethod = this.safeString (this.options, 'fetchTicker');
         const symbol = this.safeSymbol (undefined, market);
         let ask = undefined;
         let bid = undefined;
         let last = undefined;
+        let percentage = undefined;
         const timestamp = this.milliseconds ();
-        if (typeof ticker !== 'string') {
+        if (fetchTickerMethod === 'fetchTickerV2') {
```

Comment:
@Dan-krm we should avoid relying on the options here, why not use `safeNumber2` instead?

Reply:
✅